flask_app.py

```diff
@@ -232,22 +254,75 @@ def check_identity_or_auth(request):
     if access_token is None:
         log.debug(f"Incoming request with no access token, telling them to auth the app")
         link = make_authorization_url()
-        resp = make_response(render_template('auth.html', authlink=link))
+        discordlink = make_discord_authorization_url()
+        resp = make_response(render_template('auth.html', authlink=link, discordauthlink=discordlink))
         return (resp, None)
 
     headers = {"Authorization": "bearer " + access_token, 'User-agent': 'CFB Risk Orders'}
     response = requests.get(REDDIT_ACCOUNT_URI, headers=headers)
     if response.status_code == 401:
-        log.error(f"{access_token},401 Error from CFBR API")
-        link = make_authorization_url()
-        resp = make_response(render_template('auth.html', authlink=link))
-        return (resp, None)
+        headers = {"Authorization": "Bearer " + access_token, 'User-agent': 'CFB Risk Orders'}
+        response = requests.get(DISCORD_ACCOUNT_URI, headers=headers)
```

Comment:
Ohh it just tries to auth with Reddit and then Discord every time? I don't like that. Maybe we need to add another cookie that says which service the access_token is associated with.

Reply:
I don't really know how to detect which button they clicked, could you help me with that piece?

Reply:
Oh... unless we put it in the callback route? Maybe I can figure that out.